### ISSUE TITLE
AudioContext sampling rate may be different on some machines

### ISSUE THREAD

**Issue by JonBoley** [tietgenlabs/audiovisual#2]
```
Uncaught NotSupportedError: Failed to set the 'buffer' property on 'ConvolverNode': The buffer sample rate of 44100 does not match the context rate of 48000 Hz.
```

https://github.com/WebAudio/web-audio-api/issues/300

We may need to have alternate versions of HRTFs (and audio files?) for other sample rates.

**Comment by rtoy** [comment]
You don't say, but how did you get the buffer that you are using for the ConvolverNode?

I'm also not sure how that relates to the fact that the context sampling rate may be different on some machines. It can be different even on the same machine, either, because you changed the rate or output is going to different devices.

**Comment by JonBoley** [comment]
Quick reply before bed (in Europe)...
I think we need to detect the context sampling rate, then load the appropriate data into the buffer for the ConvolverNode.
Does that address your question at all?

**Comment by rtoy** [comment]
Not really.  But I guess this is an issue for you, not WebAudio per se.

**Comment by JonBoley** [comment]
The issue is that the context sampling rate may be different for different users. We really should use audio files that match the context sampling rate.

**Comment by JonBoley** [comment]
If you can load audio files based on the context, let me know and I can provide alternative way and hrtf files

**Comment by rtoy** [comment]
How do you get the buffer being used for the convolver node.  If you are using decodeAudioData, then the file should be at the correct rate for the context (by resampling).  If you create it yourself in some other way, you'll have to do the resampling yourself.

**Comment by JonBoley** [comment]
Oh, I see what you were saying.
See here for an example of resampling the HRTF data and using that in the ConvolverNode: https://github.com/JonBoley/webAudio-tests

**Comment by JonBoley** [comment]
@roykolak I just realized I wasn't talking to you in this thread... ;-)

**Comment by roykolak** [comment]
lol!
